packages/hash/frontend/src/shared/routing/route-account-info.tsx

```diff
@@ -0,0 +1,56 @@
+import { useRouter } from "next/router";
+import { createContext, VFC, useContext, useMemo, ReactNode } from "react";
+
+type RouteAccountInfo = {
+  accountId: string;
+};
+
+const RouteAccountInfoContext = createContext<RouteAccountInfo | undefined>(
+  undefined,
+);
+
+/**
+ * @todo we currently pull the accountId from the url and that works for now
+ * although this wouldn't work when we switch to using slugs instead of accountIds in the url.
+ * When that happens the accountId should be pulled properly in this component
+ */
+export const RouteAccountInfoProvider: VFC<{ children?: ReactNode }> = ({
+  children,
+}) => {
+  const router = useRouter();
+
+  const accountSlug = router.query["account-slug"];
+
+  const contextValue = useMemo<RouteAccountInfo | undefined>(
+    () =>
+      typeof accountSlug === "string"
+        ? {
+            accountId: accountSlug, // @todo parse and use suspense for lookup if needed
+          }
+        : undefined,
+    [accountSlug],
+  );
+
+  return (
+    <RouteAccountInfoContext.Provider value={contextValue}>
+      {children}
+    </RouteAccountInfoContext.Provider>
+  );
+};
+
+export interface UseRouteAccountInfo {
+  (options: { allowUndefined: true }): RouteAccountInfo | undefined;
+  (options?: { allowUndefined?: false }): RouteAccountInfo;
+}
```

Comment:
@teenoh this gives us extra type safety when we use the hook. We don’t need `!` as discussed in https://github.com/hashintel/hash/pull/460#discussion_r841612676